src/contexts/browser-context.js

```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext, useReducer } from "react";
+import { browserReducer } from "./browserReducer";
+
+const BrowserContext = createContext();
+
+const BrowserProvider = ({ children }) => {
+    const [browserState, browserDispatch] = useReducer(browserReducer, {
```

Comment:
you can destructure state values directly over here, so there is no need to destructure it every where

Reply:
will modify it